CalibCalorimetry/HcalAlgos/src/HcalPulseShapes.cc

```diff
@@ -708,7 +710,30 @@ void HcalPulseShapes::computeSiPMShape2017()
 
   for (unsigned int j = 1; j <= nBinsSiPM_; ++j) {
     nt[j] /= norm;
-    siPMShape2017_.setShapeBin(j,nt[j]);
+    siPMShapeMC2017_.setShapeBin(j,nt[j]);
+  }
+}
+
+void HcalPulseShapes::computeSiPMShapeHE206()
+{
+  //numerical convolution of SiPM pulse + WLS fiber shape
+  std::vector<double> nt = convolve(nBinsSiPM_,analyticPulseShapeSiPMHE,Y11206);
+
+  siPMShapeMC2018_.setNBin(nBinsSiPM_);
+
+  //Aligning 206 phase closer to 205 in order to have good reco agreement
+  int shift = -2;
+
+  //skip first bin, always 0
+  double norm = 0.;
+  for (int j = 1; j <= nBinsSiPM_; ++j) {
```

Comment:
hi @cmadrid1 - better to write this as 
for ( unsigned int j = std::max(1,-1*shift), j<=nBinsSiPM_; j++) {
 norm += std::min(0., nt[j-shift])
}
double normInv=1./norm;
for ( int j=1; j<=nBinsSiPM_;j++) {
  if ( j-shift>=0 ) {
     nt[j-shift] *=norm;
     siPMShapeMC2018_.setShapeBin(j,nt-shift]);
  }
  else{
    siPMShapeMC2018_.setShapeBin(j,0)
}

to avoid out of bounds access and excessive divisions.